src/rcl_handle.cpp

```diff
@@ -86,7 +86,8 @@ NAN_METHOD(RclHandle::Dismiss) {
 }
 
 v8::Local<v8::Object> RclHandle::NewInstance(void* handle, RclHandle* parent,
-                                             std::function<int()> deleter) {
+                                             std::function<int()> deleter,
```

Comment:
@koonpeng @minggangw I can see the merit to your recommendation to delegate managing ptr memory to a deleter function. I feel making such a change is beyond the scope of this PR due to the large impact it would have across the addon codebase. My preference is to continue with my original technique/hack, i.e., where my deleter function sets the rclHandle's ptr to null, and open a separate issue for how RclHandle handles free it's ptr.

Reply:
+1 We'd better focus on the lifecycle feature within this PR, considering we have already had a lot of changes. @wayneparrott please go-ahead to open an issue, as @koonpeng said we should have a completely user-defined deleter.

Reply:
Opened issue #728